### PR TITLE
feat(aio): add GH-style anchor links to headings

### LINKED ISSUE #16126: Add anchors & anchor icons to all headers
**Current behavior**
It's not possible to link to an arbitrary header in a document content.

**Expected behavior**
Github and many other sites (wikipedia?) allow you to however over any header and activate an icon that allows the reader to construct or navigate to the current page and scroll down to the selected header. Example from https://github.com/angular/angular:

screen shot 2017-04-18 at 8 58 11 pm

It would be awesome to have this feature for all of our headers, as this would allow us to completely get rid of the address bar on mobile devices, because there the anchor could be used to retrieve the url of the current page.

### PR DESCRIPTION
**Pull request by petebacondarwin**
Closes #16126

### CODE REVIEW

File: aio/src/styles/2-modules/_heading-anchors.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
...n8Ax7A5WQAAAHicY2BkYGAA4teL1+yI57f5ysDNwgAC529f0kOmWRiYVgEpDgYmEA8AUzEKsQAAAHicY2BkYGB2+O/AEMPCAAJAkpEBFbAAADgKAe0EAAAiAAAAAAQAAAAEAAAAAAAAKgAqACoAiAAAeJxjYGRgYGBhsGFgYgABEMkFhAwM/xn0QAIAD6YBhwB4nI1Ty07cMBS9QwKlQapQW3VXySvEqDCZGbGaHULiIQ1FKgjWMxknMfLEke2A+IJu+wntrt/QbVf9gG75jK577Lg8K1qQPCfnnnt8fX1NRC/pmjrk/zprC+8D7tBy9DHgBXoWfQ44Av8t4Bj4Z8CLtBL9CniJluPXASf0Lm4CXqFX8Q84dOLnMB17N4c7tBo1AS/Qi+hTwBH4rwHHwN8DXqQ30XXAS7QaLwSc0Gn8NuAVWou/gFmnjLrEaEh9GmDdDGgL3B4JsrRPDU2hTOiMSuJUIdKQQayiAth69r6akSSFqIJuA19TrzCIaY8sIoxyrNIrL//pw7A2iMygkX5vDj+G+kuoLdX4GlGK/8Lnlz6/h9MpmoO9rafrz7ILXEHHaAx95s9lsI7AHNMBWEZHULnfAXwG9/ZqdzLI08iuwRloXE8kfhXYAvE23+23DU3t626rbs8/8adv+9DWknsHp3E17oCf+Z48rvEQNZ78paYM38qfk3v/u3l3u3GXN2Dmvmvpf1Srwk3pB/VSsp512bA/GG5i2WJ7wu430yQ5K3nFGiOqgtmSB5pJVSizwaacmUZzZhXLlZTq8qGGFY2YcSkqbth6aW1tRmlaCFs2016m5qn36SbJrqosG4uMV4aP2PHBmB3tjtmgN2izkGQyLWprekbIntJFing32a5rKWCN/SdSoga45EJykyQ7asZvHQ8PTm6cslIpwyeyjbVltNikc2HTR7YKh9LBl9DADC0U/jLcBZDKrMhUBfQBvXRzLtFtjU9eNHKin0x5InTqb8lNpfKv1s1xHzTXRqgKzek/mb7nB8RZTCDhGEX3kK/8Q75AmUM/eLkfA+0Hi908Kx4eNsMgudg5GLdRD7a84npi+YxNr5i5KIbW5izXas7cHXIMAau1OueZhfj+cOcP3P8MNIWLyYOBuxL6DRylJ4cAAAB4nGNgYoAALjDJyIAOWMCiTIxMLDmZedkABtIBygAAAA==) format('woff');
```
Comment (IgorMinar) [inline aio/src/styles/2-modules/_heading-anchors.scss]: is this a free font? can you add a comment as to how we obtained it?

**Review by IgorMinar — APPROVED** [review]
other than the origin of the font this looks good

**Comment by petebacondarwin** [comment]
Unlike @gkalpak, CSS is not my forte! So I expect that this PR (or a subsequent PR) will need some styling love from @sjtrimble

**Comment by petebacondarwin** [comment]
OK, I switched it to use the material icons font.
Should be good to go.
@wardbell can you review?

**Comment by sjtrimble** [comment]
In case you want to add these:

`span` tag

- margin: 8px;

`a` tag 

- display: inline-block;
- vertical-align: middle;

**Comment by petebacondarwin** [comment]
Thanks @sjtrimble - I added those styles and it looks much better.

**Pull request closed without merging**